src/Event/Event.js

```diff
@@ -57,15 +57,16 @@ const Event = ({
     ];
     Animated.parallel(animations).start();
   }, [position]);
-  const panResponder = useRef(
-    PanResponder.create({
-      onStartShouldSetPanResponder: () => isDragEnabled(),
+
+  const panResponder = useMemo(() => {
```

Comment:
Hi @pdpino , can you explain more why do we need `useMemo` here? I have a look at the related issue but I don't really understand the root cause...

Reply:
Sure @hoangnm

* `panResponder = useRef(...).current` creates an instance of a `PanResponder` in the first render, and does not re-create this instance in the next renders
* That `panResponder` instance saves a reference to the method `onDragRelease`, which uses the `position` prop received in the first render
* --> on the first drag-and-drop the `position` used by  `onDragRelease()` is correct
  --> on the next drag-and-drops the `position` used by  `onDragRelease()` is the old one --> buggy movements
* My solution was `useMemo()` --> the `panResponder` gets recreated each time any of the values change
* Sounds kind of inefficient, but was the only solution I found

Note: the bug does not occur when dragging to a different day, because that triggers the creation of a new `Event` component (--> new `panResponder` is created)

Hope this is more clear and not more confusing :laughing:

Reply:
Thanks, @pdpino, I understand now. I'm having a concern about the performance when re-creating the `panResponder` too many times. I think another option is moving the logic that calculates the new position from `onDragRelease` to the parent component. WDYT?

Reply:
I'm also concerned about the performance, but I think moving the logic would not solve it:

* The [`onDrag()` callback](https://github.com/pdpino/react-native-week-view/blob/bug/drag-drop-memoized/src/Event/Event.js#L20) would require changing the signature to something like this: `onDrag(event, position, dx, dy)`
* --> the `panResponder` still needs to be re-created if the `position` changes, i.e.:

  ```js
  panResponder = useMemo(
    () => PanResponder.create({
      // ...
      onPanResponderRelease: (_, gestureState) => {
        const { dx, dy } = gestureState;
        onDrag(event, position, dx, dy); // Call the parent component directly
      },
      // ...
    }),
    [event, position, ...], // position must be included here
  )
  ```
  to avoid memoizing the `position` and triggering the same bug
* See the react-native-draggable library with a [similar solution with `useMemo`](https://github.com/tongyy/react-native-draggable/blob/master/Draggable.js#L141)
* Another option would be using the signature `onDrag(event, dx, dy)`, and let the `Events` component find the `position` for the `event` (given its `id`, or some identifier), instead of passing as parameter. But this would get messy and bug-prone

I think we can keep this bugfix for now, and review + improve the performance later

Reply:
Thank @pdpino, that makes sense.